daemon/state.go

```diff
@@ -11,16 +11,18 @@ import (
 
 type State struct {
 	sync.Mutex
-	Running    bool
-	Paused     bool
-	Restarting bool
-	OOMKilled  bool
-	Pid        int
-	ExitCode   int
-	Error      string // contains last known error when starting the container
-	StartedAt  time.Time
-	FinishedAt time.Time
-	waitChan   chan struct{}
+	Running           bool
+	Paused            bool
+	Restarting        bool
+	OOMKilled         bool
+	removalInProgress bool // Not need for this to be persistent on disk.
```

Comment:
@LK4D4 

Now I am thinking that it is better to keep it where it is. Reason being that right now all the possible container states are together. If I move it below waitChan, then a single state is out of the group. And I think it is better to keep all states together instead of forcing grouping based on public/private.